src/models/language_model.cc

```diff
@@ -360,9 +379,13 @@ namespace ctranslate2 {
       std::vector<StorageView> inputs{ids};
 
       if (_encoder->num_input_features() > 1) {
-        StorageView token_type_ids(ids.shape(), ids.dtype(), device);
-        token_type_ids.zero();
-        inputs.emplace_back(std::move(token_type_ids));
+        if (token_type_ids.empty()) {
+          StorageView placeholder_type_ids(ids.shape(), ids.dtype(), device);
+          placeholder_type_ids.zero();
+          inputs.emplace_back(std::move(placeholder_type_ids));
+        } else {
+          inputs.emplace_back(std::move(token_type_ids));
```

Comment:
`token_type_ids` is a const reference so the `std::move` has no effect.

```suggestion
          inputs.emplace_back(token_type_ids);
```